### ISSUE TITLE
ControlStructureSpacingSniff fails on multiline string

### ISSUE THREAD

**Issue by alpha0010** [squizlabs/PHP_CodeSniffer#2642]
Fails (with infinite loop?) when handling a multiline string.
```
/ccel-shared# cli/vendor/bin/phpcs --version                 
PHP_CodeSniffer version 3.5.0 (stable) by Squiz (http://www.squiz.net)

/ccel-shared# cli/vendor/bin/phpcs --standard=PSR12 test.php 

FILE: /ccel-shared/test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 3 | ERROR | [x] The first expression of a multi-line control
   |       |     structure must be on the line after the opening
   |       |     parenthesis
 6 | ERROR | [x] Each line in a multi-line control structure must be
   |       |     indented at least once; expected at least 4 spaces,
   |       |     but found 0
 6 | ERROR | [x] The closing parenthesis of a multi-line control
   |       |     structure must be on the line after the last
   |       |     expression
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 72ms; Memory: 6MB

/ccel-shared# cli/vendor/bin/phpcbf -v --standard=PSR12 test.php 
Registering sniffs in the PSR12 standard... DONE (59 sniffs registered)
Creating file list... DONE (1 files in queue)
Changing into directory /ccel-shared
Processing test.php [PHP => 31 tokens in 8 lines]... DONE in 3ms (3 fixable violations)
        => Fixing file: 1/3 violations remaining [made 50 passes]... ERROR in 228ms

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
/ccel-shared/test.php                                 FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 2 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 303ms; Memory: 6MB
```

```php
<?php

if (preg_match("/
                  foo # The first part.
                  bar # the second part.
                /x", "foobar") === 1) {
    echo "success\n";
}
```

**Comment by jrfnl** [comment]
Duplicate of #2623 ? (already fixed)
